BiconeEvolution/current_antenna_evo_build/XF_Loop/Evolutionary_Loop/Antenna_Performance_Metric/XFintoPUEO.py

```diff
@@ -0,0 +1,345 @@
+
+#numFreq = 130 # 1500 - 200 = 1300 / 10 = 130 steps    
+
+freqVals = [83.33*10**(+6), 100.00*10**(+6), 116.67*10**(+6), 133.33*10**(+6), 150.00*10**(+6), 166.67*10**(+6), 183.34*10**(+6), 200.00*10**(+6), 216.67*10**(+6), 233.34*10**(+6), 250.00*10**(+6), 266.67*10**(+6), 283.34*10**(+6), 300.00*10**(+6), 316.67*10**(+6), 333.34*10**(+6), 350.00*10**(+6), 366.67*10**(+6), 383.34*10**(+6), 400.01*10**(+6), 416.67*10**(+6), 433.34*10**(+6), 450.01*10**(+6), 466.67*10**(+6), 483.34*10**(+6), 500.01*10**(+6), 516.68*10**(+6), 533.34*10**(+6), 550.01*10**(+6), 566.68*10**(+6), 583.34*10**(+6), 600.01*10**(+6), 616.68*10**(+6), 633.34*10**(+6), 650.01*10**(+6), 666.68*10**(+6), 683.35*10**(+6), 700.01*10**(+6), 716.68*10**(+6), 733.35*10**(+6), 750.01*10**(+6), 766.68*10**(+6), 783.35*10**(+6), 800.01*10**(+6), 816.68*10**(+6), 833.35*10**(+6), 850.02*10**(+6), 866.68*10**(+6),883.35*10**(+6), 900.02*10**(+6), 916.68*10**(+6), 933.35*10**(+6), 950.02*10**(+6), 966.68*10**(+6), 983.35*10**(+6), 1000.00*10**(+6), 1016.70*10**(+6), 1033.40*10**(+6), 1050.00*10**(+6), 1066.70*10**(+6)]
+
+numFreq = 60
+
+parser = argparse.ArgumentParser()
+parser.add_argument("NPOP", help="How many antennas are being simulated each generation?", type=int)
+parser.add_argument("WorkingDir", help="Evolutionary_Loop directory", type=str)
+parser.add_argument("RunName", help="Run Name directory where everything will be stored", type=str)
+parser.add_argument("gen", help="The generation the loop is on", type=int)
+parser.add_argument("indiv", help="The individual in the population", type=int)
+g = parser.parse_args()
+
+
+uanLoc = '/users/PAS1977/jacobweiler/GENETIS/'
+
+def readFile(indiv, freqNum):
+    uanName = f'{g.WorkingDir}/{g.RunName}/uan_files/{g.gen}_uan_files/{indiv}/{g.gen}_{indiv}_{freqNum}.uan'
+    #pulling data from the uan file                                                                                         
+    Data = np.genfromtxt(uanName, unpack=True, skip_header = 17)
+    return (Data[2]) #returns Gain(Theta) column                                                                               
+
+# Testing for Max
+### START ###
+
+n = 37
+m = 73
+
+freqArray = []
+# Formatting Frequency values into Scientific Notation  
+for q in range(numFreq):
+    scinoteval = "{:.2e}".format(freqVals[q])
+    freqArray.append(scinoteval)
```

Comment:
Using `for i in range(seq): seq[i]` is an anti-pattern, since you can just do `for val in seq:`.  Here you can also just use a list comprehension.  (And technically in Python they are called `list`s, not arrays.)

```suggestion
# Formatting Frequency values into Scientific Notation
freqArray = [f'{val:.2e}' for val in freqVals]
```

This should be equivalent, assuming that `numFreq == freqVals`, if not:
```suggestion
# Formatting Frequency values into Scientific Notation
freqArray = [f'{freqVals[n]:.2e}' for n in range(numFreq)]
```